processor/src/main/java/org/mapstruct/ap/internal/model/NestedTargetPropertyMappingHolder.java

```diff
@@ -180,9 +180,15 @@ public NestedTargetPropertyMappingHolder build() {
                         .groupedBySourceReferences
                         .entrySet() ) {
                         PropertyEntry sourceEntry = entryBySP.getKey();
+                        boolean forceUpdateMethodOrNonNestedReferencesPresent =
+                            forceUpdateMethod || !groupedSourceReferences.nonNested.isEmpty();
+                        // if an update method is forced or there are non nested source references then
```

Comment:
Q1. The only places were we currently fall back on update method is when we have 2 source arguments in the mapping method right?

Q2. This is a new situation. I think we need to explain what actually triggers this update methods. There have been many discussions in the past why we use them in a chain that only uses create methods.

I'm not sure.. In general. were would this lead is in the future? Should we actually allow this situation?

Reply:
Q1: Yes until now the only place where we fall back to update method was having 2 source methods.

Q2: Exactly this is another corner case. All these cases are quite complex and are forcing us to do some complex logic in there.

I agree with you, this can be more problematic in the future. 

What we could do is not to allow mappings that are not consistent mappings to different sources should not be allowed maybe. This means that we would have to do checks at the beginning and if we see something like this we error out. So if we have not symmetric mappings to sources (mappings to different sources within the levels)

Reply:
> What we could do is not to allow mappings that are not consistent mappings to different sources should not be allowed maybe.

I'm just thinking: what are frameworks like dozer doing in these cases? Are we missing something?

Anyway.. If we can support a case in a simple manner (like above) we probably should.

Reply:
I am not entirely sure what the other frameworks do in such a case. Maybe we need to test it 😄